### PR TITLE
New axis lazy pt2

### PR DESCRIPTION
**Pull request by pelson**
Update `iris.util.new_axis` and `cube.transpose` (which is barely tested directly) to be lazy.

### CODE REVIEW

File: lib/iris/cube.py [hunk 1]
```diff
@@ -1521,6 +1521,11 @@ def data(self, value):
         self._my_data = data
 
     def has_lazy_data(self):
+        """
+        Return whether the underlying data of this cube is a biggus array
+        (and therefore potentially unloaded from disk).
```
Comment (rhattersley) [inline lib/iris/cube.py]: Not a fan of "unloaded" - perhaps "has not been loaded"?

File: docs/iris/src/whatsnew/contributions_1.9/newfeature_2015-10-19_new_axis_lazy.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+iris.util.new_axis and cube.transpose can now operate on lazy data
```
Comment (rhattersley) [inline docs/iris/src/whatsnew/contributions_1.9/newfeature_2015-10-19_new_axis_lazy.txt]: This could do with a leading `* ` to make it a bullet point.

**Comment by pelson** [comment]
Commit added.

**Comment by pelson** [comment]
The big move to dask for Iris v2 made transpose and new_axis lazy. :tada:

**Pull request closed without merging**